pytroll_collectors/segments.py

```diff
@@ -851,10 +850,32 @@ def _add_existing_files_to_slot(self, slot, fnames, message):
 
 def _get_existing_files_from_message(message):
     mask = message.pattern.parser.globify({})
-    path = urlparse(message.message_data["uri"]).path
-    base_dir = os.path.dirname(path)
+    url_parts = urlparse(message.message_data["uri"])
 
-    return glob.glob(os.path.join(base_dir, mask))
+    return _fsspec_glob(url_parts, mask)
+
+
+def _fsspec_glob(url_parts, mask):
+    import fsspec
+
+    pattern = urlunparse(
+        (
+            url_parts.scheme,
+            url_parts.netloc,
+            '/'.join(url_parts.path.split('/')[:-1]) + '/' + mask,
+            '',
+            '',
+            ''
+        )
+    )
+
+    fs_ = fsspec.filesystem(url_parts.scheme)
+    files = fs_.glob(pattern)
+    # There might be no scheme in the returned filenames, so add it if scheme is defined
+    if url_parts.scheme:
+        files = [url_parts.scheme + '://' + f for f in files if not f.startswith(
```

Comment:
Wouldn't `urlunparse` be preferable here?

Reply:
Using `urlunparse()` the one-liner would be:
```python
    if url_parts.scheme:
        files_with_scheme = []
        for f in files:
            parts = list(urlparse(f))
            parts[0] = url_parts.scheme
            # Netloc is empty, so we need to add the scheme slashes manually
            parts[1] = "//"
            files_with_scheme.append(urlunparse(parts))
        files = files_with_scheme
    return files
```
Not an improvement in my opinion.

Reply:
indeed, it doesn't look better. It's just the ` + "://" + ` I wasn't so happy about, but we can leave it be.

Reply:
Oh, and this alternate version would break if there were a `netloc` in the URI.